File: samples/Mathematica/sorting-algorithms-insertion-sort.m
insertionSort[a_List] := Module[{A = a},
  For[i = 2, i <= Length[A], i++,
   value = A[[i]];    j = i - 1;
   While[j >= 1 && A[[j]] > value, A[[j + 1]] = A[[j]]; j--;];
   A[[j + 1]] = value;];
A
]

File: samples/JavaScript/sorting-algorithms-insertion-sort.js
function insertionSort (a) {
    for (var i = 0; i < a.length; i++) {
        var k = a[i];
        for (var j = i; j > 0 && k < a[j - 1]; j--)
            a[j] = a[j - 1];
        a[j] = k;
    }
    return a;
}

var a = [4, 65, 2, -31, 0, 99, 83, 782, 1];
insertionSort(a);
document.write(a.join(" "));
